code/game/machinery/machinery.dm

```diff
@@ -351,3 +351,37 @@ Class Procs:
 	icon_state = "autolathe"
 	density = TRUE
 	anchored = TRUE
+
+//fuel pomp
+/obj/structure/machinery/fuelpump
+	name = "\improper Fuel Pump"
+	layer = ABOVE_MOB_LAYER
+	desc = "It is a machine that pump fuel in lifeboats."
+	icon = 'icons/obj/structures/machinery/fuelpump.dmi'
+	icon_state = "fuelpump_off"
+	density = TRUE
+	anchored = TRUE
+
+
+/obj/structure/machinery/fuelpump/update_icon()
+	if(stat & NOPOWER)
+		icon_state = "fuelpump_off"
+	else
+		icon_state = "fuelpump_on"
+
+		//if fuel pump 0
+		if(fuelpump_stage == 1)
+			icon_state = "fuelpump_0"
+		//if fuel pump 25
+		if(fuelpump_stage == 25)
+			icon_state = "fuelpump_25"
+		//if fuel pump 50
+		if(fuelpump_stage == 50)
+			icon_state = "fuelpump_50"
+		//if fuel pump 75
+		if(fuelpump_stage == 75)
+			icon_state = "fuelpump_75"
+		//if fuel pump 100
+		if(fuelpump_stage == 100)
+			icon_state = "fuelpump_100"
+//currently doesn't update properly
```

Comment:
switch statement maybe?

Reply:
you know what u could do actually, if u changed the dmi iconstate from fuelpump_0 to fuelpump_1 
you could maybe just do this, should work I think.
```
if(fuelpump_stage >= 1)
      icon_state="fuelpump_[fuelpump_stage]"
```